core/reflection.jvm/src/kotlin/reflect/jvm/internal/KPropertyImpl.kt

```diff
@@ -188,6 +214,9 @@ internal abstract class KPropertyImpl<out V> private constructor(
 
     companion object {
         val EXTENSION_PROPERTY_DELEGATE = Any()
+
+        private fun Any?.defaultIfExtension(type: Class<*>) =
+            if (this === EXTENSION_PROPERTY_DELEGATE) ReflectedArray.get(ReflectedArray.newInstance(type, 1), 0) else this
```

Comment:
Please use `kotlin.reflect.jvm.internal.defaultPrimitiveValue` instead

(for the record, tricks like this lead to issues in Graal VM native image like https://youtrack.jetbrains.com/issue/KT-44594.)

Reply:
Thanks, I thought that this should already exist but missed it somehow.